src/pages/index.tsx

```diff
@@ -1,21 +1,79 @@
 import styles from "../styles/Page.module.scss";
-import { KeywordRadioBtnGroup } from "@/components/KeywordRadioBtnGroup/KeywordRadioBtnGroup";
+import {
+  KeywordRadioBtnGroup,
+  SearchKeywords,
+} from "@/components/KeywordRadioBtnGroup/KeywordRadioBtnGroup";
 import { ModeToggleBtn } from "@/components/ModeToggleBtn/ModeToggleBtn";
 import { Logo } from "@/components/Logo/Logo";
 import { Searchbar } from "@/components/Searchbar/Searchbar";
+import { ResultCards } from "@/components/ResultCards/ResultCards";
 import { useState, useCallback } from "react";
 
+const baseEndpoint = "https://api.openbrewerydb.org/v1/breweries";
+
 export default function Home() {
   const [keyword, setKeyword] = useState("name");
-  const [text, setText] = useState("");
-  const [submitted, setSubmitted] = useState("");
+  const [query, setQuery] = useState("");
+  const [queryResults, setQueryResults] = useState([]);
+  const [shouldShowEmpty, setShouldShowEmpty] = useState(true);
+
+  const fetchBreweries = useCallback(
+    async (url: string, keyword: string) => {
+      const response = await fetch(url);
+
+      if (keyword === SearchKeywords.country) {
+        const unfilteredData = await response.json();
+        return unfilteredData.filter((breweryData: any) =>
+          breweryData.country.toLowerCase().includes(query.toLowerCase())
+        );
+      }
+
+      return await response.json();
```

Comment:
you have duplication of `await response.json();`

instead of this first read the response, and than return apropriate data

```
const data = await response.json();

if (keyword === SearchKeywords.country) {
  retrun data.filter...
}

return data

Reply:
fixed